interfaces/core.go

```diff
@@ -157,6 +157,18 @@ type Interface interface {
 	AutoConnect(plug *Plug, slot *Slot) bool
 }
 
+// Recorder describes interactions between backends and interfaces.
```

Comment:
I think it makes sense. I've changed everything to follow this and reading the diff it looks good. Have a look please.